src/index.ts

```diff
@@ -1,63 +1,133 @@
-import Capture from './Capture'
-import Auth from './Auth'
-import License from './License'
-import KeyService from './Key'
-import Utils from './utils'
-import { Photo } from '@capacitor/camera'
-
-
-export default class TikiClient{
-  private keyService = new KeyService()
-  private address: string = '';
-  public capture = new Capture()
-  public auth = new Auth(this.keyService)
-  public license = new License()
-
-  public async initialize(providerId: string, pubKey: string, userId: string){
-
-    await this.keyService.clear()
-
-    const keys = await this.keyService.get()
-
-    if(keys.find((key)=> key.value.name === `${providerId}.${userId}`)) return 
-
-    const address = await this.auth.registerAddress(providerId, pubKey, userId)
-
-    this.address = address
+import Capture from "./Capture";
+import Auth from "./Auth";
+import License from "./License";
+import KeyService from "./Key";
+import Utils from "./utils";
+import { Photo } from "@capacitor/camera";
+import { SavedKey } from "./Key/types";
+import type { PostGuardRequest, RspGuard, PostLicenseRequest} from "./License/types";
+
+export default class TikiClient {
+  private static keyService = new KeyService();
+
+  public static capture = new Capture();
+  public static auth = new Auth(TikiClient.keyService);
+  public static license = new License();
+
+  private constructor() {}
+
+  /**
+   * Initialize the TikiClient and register the device's address.
+   * @param {string} providerId - the provider ID of the associated provider account.
+   * @param {string} pubKey - the public key of the created provider.
+   * @param {string} userId - the ID to be registered to identify the user.
+   */
+  public static async initialize(
```

Comment:
providerId and pubKey should be env vars. 
Remove the parameters from the initialize method.